Source/Models/EThreeParams.swift

```diff
@@ -0,0 +1,92 @@
+//     notice, this list of conditions and the following disclaimer.
+//
+//     (2) Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in
+//     the documentation and/or other materials provided with the
+//     distribution.
+//
+//     (3) Neither the name of the copyright holder nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ''AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Lead Maintainer: Virgil Security Inc. <support@virgilsecurity.com>
+//
+
+import VirgilSDK
+
+/// Contains parameters for initializing EThree
+/// - Tag: EThreeParams
+@objc(VTEEThreeParams) public class EThreeParams: NSObject {
+    /// Identity of user
+    @objc public let identity: String
+    /// Callback to get Virgil access token
+    @objc public let tokenCallback: EThree.RenewJwtCallback
+    /// [ChangedKeyDelegate](x-source-tag://ChangedKeyDelegate) to notify changing of User's keys
+    @objc public weak var changedKeyDelegate: ChangedKeyDelegate? = nil
+    /// `KeychainStorageParams` with specific parameters
+    @objc public var storageParams: KeychainStorageParams? = nil
+
+    @objc public var enableRatchet: Bool = Defaults.enableRatchet
+
+    @objc public var keyRotationInterval: TimeInterval = Defaults.keyRotationInterval
+
+    private struct BiometryConfig: Decodable {
+        var enableRatchet: Bool = Defaults.enableRatchet
+        var keyRotationInterval: TimeInterval = Defaults.keyRotationInterval
+    }
+
+    /// Initializer with parameters from config plist file
+    ///
+    /// - Parameters:
+    ///   - identity: Identity of user
+    ///   - tokenCallback: Callback to get Virgil access token
+    ///   - configUrl: URL of config file
+    /// - Throws: corresponding error
+    @objc public convenience init(identity: String,
+                                  tokenCallback: @escaping EThree.RenewJwtCallback,
+                                  configUrl: URL) throws {
+        let data = try Data(contentsOf: configUrl)
+
+        let config = try PropertyListDecoder().decode(BiometryConfig.self, from: data)
```

Comment:
What happens if some of parameters are absent in config?

Reply:
Error is thrown. If we set default value for missing parameters, user might have grammar mistake and never know about that, till he write test on that.

Reply:
This is unacceptable, because adding new possible arguments to config will break backward compatibility. Another way to avoid typos in config is to check for unused values

Reply:
Added decoding with default parameters and check on unknown keys in config